api/src/main.ts

```diff
@@ -105,11 +104,28 @@ async function bootstrap() {
     app.useWebSocketAdapter(redisIoAdapter);
   }
 
-  process.on('uncaughtException', async (error) => {
-    if (error.stack?.toLowerCase().includes('smtp')) {
-      const logger = await app.resolve(LoggerService);
-      logger.error('SMTP error', error.stack);
-    } else throw error;
+  /**
+   * Global handler for uncaught exceptions.
+   *
+   * This handler logs all uncaught exceptions directly to stderr (console.error)
+   * with a timestamp and the error stack (or error object if no stack is present).
+   *
+   * The log format is:
+   *   [timestamp] Uncaught Exception <error stack or error>
+   *
+   * After logging, the process always exits with code 1 to avoid running in an undefined state.
+   *
+   * NOTE: Do NOT include async calls inside this handler. The Node.js process may be in an unstable state,
+   * and async operations are discouraged and may not complete as expected.
+   *
+   * @see https://nodejs.org/api/process.html#event-uncaughtexception
+   */
+  process.on('uncaughtException', (error) => {
+    const timestamp = new Date().toISOString();
+    // eslint-disable-next-line no-console
+    console.error(`[${timestamp}] Uncaught Exception`, error.stack || error);
```

Comment:
LoggerService extends from ConsoleLogger, which is a class that you can instanciate directly. Can you try that ?